tensorboard/writer/event_file_writer.py

```diff
@@ -0,0 +1,228 @@
+        disk via the add_event method.
+        The other arguments to the constructor control the asynchronous writes to
+        the event file:
+
+        Args:
+          logdir: A string. Directory where event file will be written.
+          max_queue: Integer. Size of the queue for pending events and summaries.
+          flush_secs: Number. How often, in seconds, to flush the
+            pending events and summaries to disk.
+        """
+        self._logdir = logdir
+        self._event_queue = six.moves.queue.Queue(max_queue)
+        self._ev_writer = EventsWriter(os.path.join(
+            self._logdir, "events"), filename_suffix)
+        self._flush_secs = flush_secs
+        self._closed = False
+        self._worker = _EventLoggerThread(self._event_queue, self._ev_writer,
+                                          flush_secs)
+
+        self._worker.start()
+
+    def get_logdir(self):
+        """Returns the directory where event file will be written."""
+        return self._logdir
+
+    def reopen(self):
+        """Reopens the EventFileWriter.
+
+        Can be called after `close()` to add more events in the same directory.
+        The events will go into a new events file and a new write/flush worker
+        is created. Does nothing if the EventFileWriter was not closed.
+        """
+        if self._closed:
+            self._closed = False
+            self._worker = _EventLoggerThread(
+                self._event_queue, self._ev_writer, self._flush_secs
+            )
+            self._worker.start()
+
+    def add_event(self, event):
+        """Adds an event to the event file.
+
+        Args:
+          event: An `Event` protocol buffer.
+        """
+        if not self._closed:
+            self._event_queue.put(event)
+
+    def flush(self):
+        """Flushes the event file to disk.
+
+        Call this method to make sure that all pending events have been written to
+        disk.
+        """
+        if not self._closed:
+            self._event_queue.join()
+            self._ev_writer.flush()
+
+    def close(self):
+        """Performs a final flush of the event file to disk, stops the
+        write/flush worker and closes the file. Call this method when you do not
+        need the summary writer anymore.
+        """
+        if not self._closed:
+            self.flush()
+            self._worker.stop()
+            self._ev_writer.close()
+            self._closed = True
+
+
+class _EventLoggerThread(threading.Thread):
+    """Thread that logs events."""
+
+    def __init__(self, queue, ev_writer, flush_secs):
+        """Creates an _EventLoggerThread.
+        Args:
+          queue: A Queue from which to dequeue events.
+          ev_writer: An event writer. Used to log brain events for
+           the visualizer.
+          flush_secs: How often, in seconds, to flush the
+            pending file to disk.
+        """
+        threading.Thread.__init__(self)
+        self.daemon = True
```

Comment:
Nit - can we call this `self._daemon` to be consistent?

Reply:
`daemon` is defined in the parent class: `threading.Thread`, so I think we can't.